lib/exprengine.cpp

```diff
@@ -2497,17 +2544,11 @@ static std::string execute(const Token *start, const Token *end, Data &data)
             const Token *cond = tok->next()->astOperand2(); // TODO: C++17 condition
             const ExprEngine::ValuePtr condValue = executeExpression(cond, data);
 
-            bool alwaysFalse = false;
-            bool alwaysTrue = false;
-            if (auto b = std::dynamic_pointer_cast<ExprEngine::BinOpResult>(condValue)) {
-                alwaysFalse = !b->isTrue(&data);
-                alwaysTrue = !alwaysFalse && !b->isEqual(&data, 0);
```

Comment:
for your information the `!alwaysFalse` here was only a short circuit.. if the flag alwaysFalse is already set there is no need to go and ask z3..